### ISSUE TITLE
A propositional memory model

### ISSUE THREAD

**Issue by YaZko** [vellvm/vellvm#110]
The current memory model collapses the non-determinism stemming from under-defined values so that it concretely works over bit representations of `dvalue`.

We could have a more abstract one that process `uvalue`. The former would be a refinement of the latter in the appropriate sense.

**Comment by Chobbes** [comment]
Initially I had thought that this would be a pain for hooking up something like Cerberus, which really has no idea about vellvm's uvalues, but I think it's actually perfectly reasonable to concretize when talking to Cerberus.

This is actually a real issue right now too, because this is a place where the semantics of vellvm differs from actual LLVM. As far as I know LLVM does not force undefined values to be concretized prior to a store, but we do.

So, in summary, yes, I think we should look into doing this :).

**Comment by YaZko** [comment]
Closing as outdated. A solution has been described in our ICFP'24 paper.
